src/utils.cpp

```diff
@@ -1114,10 +1128,20 @@ FuncsModulesMap parse_traceable_funcs()
   FuncsModulesMap result;
   std::string line;
   while (std::getline(available_funs, line)) {
+    bool skip = false;
     auto func_mod = split_symbol_module(line);
     if (func_mod.second.empty())
       func_mod.second = "vmlinux";
-    result[func_mod.first].insert(func_mod.second);
+    if (bad_funcs.find(func_mod.first) != bad_funcs.end())
+      continue;
+    for (const auto &s : bad_funcs_partial) {
+      if (!std::strncmp(func_mod.first.c_str(), s.c_str(), s.length())) {
+        skip = true;
+        break;
+      }
+    }
```

Comment:
Would be nice to enclose this check in a function (e.g. `is_bad_func`).

Reply:
I agree but this is not a deal breaker for me and having utility functions only called from one place also has a bit of smell (though it would make it easier to read).

Reply:
> Would be nice to enclose this check in a function (e.g. `is_bad_func`).

You're right @viktormalik (and @jordalgo :-) ) - it does look better with the code hoisted out into a separate function.